drivers/eeprom/include/eeprom_types.h

```diff
@@ -0,0 +1,139 @@
+     *
+     * @retval  0       Failed to initialize the device
+     * @return          The size of the device in bytes
+     */
+    eeprom_off_t (*init)(void *handle, const void *params);
+    /**
+     * @brief   Read data from the EEPROM device
+     *
+     * @param[in,out]   handle  Device handle of the EEPROM to read from
+     * @param[in]       pos     Position to start to read from (in bytes)
+     * @param[out]      dest    Buffer to write the data to
+     * @param[in]       len     Number of bytes to read
+     *
+     * @retval  0       Success
+     * @retval  -EIO    I/O error occurred while reading
+     */
+    int (*read)(void *handle, eeprom_off_t pos, void *dest, size_t len);
+    /**
+     * @brief   Write data from the EEPROM device
+     *
+     * @param[in,out]   handle  Device handle of the EEPROM to write to
+     * @param[in]       pos     Position to start to write to (in bytes)
+     * @param[in]       data    Buffer to write to the EEPROM device
+     * @param[in,out]   len     Number of bytes to write
+     *
+     * @retval  0       Success
+     * @retval  -EIO    I/O error occurred while writen
+     *
+     * If `-EFAULT` has been returned, the content of the EEPROM device is
+     * unchanged. If `0` is returned, all data has been written successfully
+     * to the device. If `-EIO` is returned, the contents in `[pos; pos + len[`
+     * of the device are undefined, but the remaining content of the device
+     * is unchanged.
+     */
+    int (*write)(void *handle, eeprom_off_t pos, const void *data, size_t len);
+    /**
+     * @brief   Erase the whole EEPROM
+     *
+     * @param[in,out]   handle  Device handle of the EEPROM to erase
+     *
+     * @retval  0       Success
+     * @retval  -EIO    I/O error occurred during erase
+     *
+     * If `0` is returned a subsequent read for any valid address must return
+     * @ref eeprom_driver_t::clear_byte. If `-EIO` is returned, the whole
+     * content of the EEPROM is undefined.
+     *
+     * This function is optional to implement by the driver. If the function
+     * pointer is `NULL`, a fallback implementation that calls @ref eeprom_set
+     * is used instead. This will fill the EEPROM with
+     * @ref eeprom_driver_t::clear_byte.
+     */
+    int (*erase)(void *handle);
+    /**
+     * @brief   The EEPROM is filled with this byte after being erased
+     */
+    uint8_t clear_byte;
+} eeprom_driver_t;
+
+/**
+ * @brief   The current state of an EEPROM device
+ */
+typedef struct {
+    const eeprom_driver_t *driver;  /**< The driver of the EEPROM device */
+    void *handle;                   /**< The handle to pass to the driver */
+    eeprom_off_t size;              /**< Size of the EEPROM device in bytes */
+} eeprom_dev_t;
+
+/**
+ * @brief   Initialize all EEPROM devices
+ *
+ * This function is to be called only once and during boot.
+ */
+void eeprom_init(void);
```

Comment:
Not needed anymore.

Reply:
Indeed. And additionally it was in the wrong header.

Thanks for spotting this. I removed it.